homeassistant/components/linear_garage_door/light.py

```diff
@@ -0,0 +1,141 @@
+    """Light for Linear devices."""
+
+    _attr_color_mode = ColorMode.BRIGHTNESS
+    _attr_supported_color_modes = {ColorMode.BRIGHTNESS}
+
+    def __init__(
+        self,
+        device_id: str,
+        device_name: str,
+        subdevice: str,
+        config_entry: ConfigEntry,
+        coordinator: LinearUpdateCoordinator,
+    ) -> None:
+        """Initialize the light."""
+        super().__init__(coordinator)
+
+        self._attr_has_entity_name = True
+        self._attr_name = subdevice
+        self._attr_unique_id = f"{device_id}-{subdevice}"
+        self._config_entry = config_entry
+        self._device_id = device_id
+        self._device_name = device_name
+        self._subdevice = subdevice
+
+    @property
+    def device_info(self) -> DeviceInfo:
+        """Return device info of a light."""
+        return DeviceInfo(
+            identifiers={(DOMAIN, self._device_id)},
+            name=self._device_name,
+            manufacturer="Linear",
+            model="Garage Door Opener",
+        )
+
+    @property
+    def is_on(self) -> bool:
+        """Return if the light is on or not."""
+        return bool(
+            self.coordinator.data[self._device_id]["subdevices"][self._subdevice][
+                "On_B"
+            ]
+            == "true"
+        )
+
+    @property
+    def brightness(self) -> int | None:
+        """Return the brightness of the light."""
+        return int(
+            int(
+                self.coordinator.data[self._device_id]["subdevices"][self._subdevice][
+                    "On_P"
+                ]
+            )
+            / 100
+            * 255
+        )
+
+    async def async_turn_on(self, **kwargs: Any) -> None:
+        """Turn on the light."""
+
+        linear = Linear()
+        await linear.login(
+            email=self._config_entry.data["email"],
+            password=self._config_entry.data["password"],
+            device_id=self._config_entry.data["device_id"],
+            client_session=async_get_clientsession(self.hass),
+        )
+
+        if not kwargs:
+            await linear.operate_device(self._device_id, self._subdevice, "On")
+        elif ATTR_BRIGHTNESS in kwargs:
+            brightness = str(math.floor((kwargs[ATTR_BRIGHTNESS] / 255) * 100))
+            await linear.operate_device(
+                self._device_id, self._subdevice, f"DimPercent:{brightness}"
+            )
+
+        await linear.close()
+
+    async def async_turn_off(self, **kwargs: Any) -> None:
+        """Turn off the light."""
+
+        linear = Linear()
+        await linear.login(
+            email=self._config_entry.data["email"],
+            password=self._config_entry.data["password"],
+            device_id=self._config_entry.data["device_id"],
+            client_session=async_get_clientsession(self.hass),
+        )
+        await linear.operate_device(self._device_id, self._subdevice, "Off")
+        await linear.close()
```

Comment:
I would suggest to extract the login etc. to a helper and reuse in `cover`.

Reply:
I don't think this will help much, it wouldn't really reduce on lines since we have to pass in email, password, etc. into the helper again anyways.